### PR TITLE
Feature/18329 foundation apps framework forms refactor timelines

### PR DESCRIPTION
**Pull request by ulferts**
Introduces "form--grouping" to address having multiple fields belonging together with the requirements of them being included in the table of lable - value pairs our forms usually consists of.

I tried to use fieldsets for that but apparently, fieldsets and their legend can not be styled to the desired extend. There actually seems to be a bug in at least Chrome and FF when it comes to that. 

I therefore took note of the aria attributes that are necessary to create something similar. These are part of the proposed styleguide change.

Lastly, I addressed the bug with the missing html_safe when a field had no label.

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://feature-18329-foundation-apps-framework-forms-refactor-timeline.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://feature-18329-foundation-apps-framework-forms-refactor-timeline.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

File: app/assets/stylesheets/content/_forms.md [hunk 1]
```diff
@@ -234,6 +234,45 @@
         One never lies about one's age.
       </div>
     </div>
+    <div class="form--grouping" role="group" aria-labelledby="form-grouping-label">
+        <div id="form-grouping-label" class="form--grouping-label">Colors:</div>
```
Comment (myabc) [inline app/assets/stylesheets/content/_forms.md]: @ulferts I think a 2-space indent is better.

Comment (ulferts) [inline app/assets/stylesheets/content/_forms.md]: @myabc right of course, thanks for spotting.

File: app/assets/stylesheets/content/_forms.sass [hunk 2]
```diff
@@ -350,6 +350,25 @@ fieldset.form--fieldset
...
     margin-bottom: 0.5rem
 
+.form--grouping
+  @include grid-block($wrap: true)
+  align-items: center
+
+  .form--grouping-row
+    @include grid-block(10)
+
+  .form--grouping-row+.form--grouping-row
```
Comment (myabc) [inline app/assets/stylesheets/content/_forms.sass]: @ulferts style-wise I prefer a space here.

**Comment by myabc** [comment]
> I therefore took note of the aria attributes that are necessary to create something similar. These are part of the proposed styleguide change.

@ulferts we should make a mental note to include this information for other components.

File: app/assets/stylesheets/content/_forms.md [hunk 3]
```diff
@@ -234,6 +234,46 @@
         One never lies about one's age.
       </div>
     </div>
+    <div class="form--grouping" role="group" aria-labelledby="form-grouping-label">
+      <div id="form-grouping-label" class="form--grouping-label">Colors:</div>
+        <div class="form--grouping-row">
```
Comment (myabc) [inline app/assets/stylesheets/content/_forms.md]: @ulferts indent

**Pull request merged**